src/ast.rs

```diff
@@ -201,21 +208,39 @@ impl fmt::Display for Diagnostic {
   }
 }
 
-pub(crate) struct ParsedModule {
-  comments: SingleThreadedComments,
-  module: Module,
-  text_lines: TextLines,
-}
+/// A parsed module with comments.
+pub trait ParsedAst {
+  /// Gets the comments found in the source file.
+  fn comments(&self) -> &dyn swc_common::comments::Comments;
+  /// Gets the parsed module—AST.
+  fn module(&self) -> &swc_ecmascript::ast::Module;
+  /// Gets a position from the specified byte position.
+  /// Note: This should/will panic when the byte position is
+  // outside the bounds of the source file.
+  fn get_position(&self, pos: BytePos) -> Position;
+
+  /// Get the module's leading comments, where triple slash directives might
+  /// be located.
+  fn get_leading_comments(&self) -> Vec<Comment> {
+    self
+      .comments()
+      .get_leading(self.module().span.lo)
+      .unwrap_or_else(Vec::new)
+  }
```

Comment:
If we'll make use of `ParsedAst` trait in deno_lint as well, can we also add a method to this trait that returns `MediaType` passed when calling `AstParser::parse`?
It will be useful for deno_lint to skip running some of the rules or to change the behavior of rules, depending on the MediaType. (For example, currently `no-unused-vars` identifies what media type the target file is from the file extension. But it would be great to get precise `MediaType`.)

Reply:
Done. Also added `source()` and `specifier()` methods.